### DISCUSSION TITLE
adding where clause to router get

### DISCUSSION THREAD

**Discussion by quadrastreet** [leafsphp/leaf#73] [Q&A] [answered]
I need to add this "where clause" to my route:

`where("search",".*")`

This is so that my search can contain forward slashes, such as:

`/infoSearch/forward%20slash%20is%20/`

My route call looks like this:

`$app->get("/infoSearch/{search}","InfoController@infoSearch");`

I'd like to change it to something like this:

`$app->get("/infoSearch/{search}","InfoController@infoSearch")->where("search",".*");`

But that doesn't work.  I get:

`Call to a member function where() on null`

In Laravel, I _think_ I'd be able to do this:

`Route::get("/infoSearch/{search}","InfoController@infoSearch")->where("search",".*")`

**Comment by quadrastreet** [comment]
Most likely I'll just put the string that may contain a forward slash into a query parameter... Still, I'd like to know how to add a Where clause to the route specification.

**Comment by mychidarko** [comment, marked as answer]
Leaf router does not use `where` to handle regex, it has it's own system for handling operations like this. You can find the documentation [here](https://leafphp.netlify.app/#/leaf/v/2.5.0/routing/dynamic?id=%f0%9f%a7%90-pcre-based-params)

**Reply by mychidarko** [reply]
Leaf has very extensive documentation that covers just about every feature leaf offers with links to external reads, I suggest you read through if you have any issue at all. The answer most likely lies there🥰

**Reply by quadrastreet** [reply]
I've read through tit quite a bit, but as you noted... there's a lot of documentation.

**Reply by mychidarko** [reply]
![image](https://user-images.githubusercontent.com/26604242/139143667-f9947d34-0ace-4630-953d-c8777cbf7696.png)

You'll need to read the section I linked
![image](https://user-images.githubusercontent.com/26604242/139143783-886867e3-c859-4e07-a35e-8f2f98ad9e0b.png)

So you will have this instead

```php
$app->get("/infoSearch/(.*)","InfoController@infoSearch");
```
